### PR TITLE
New Slimefun block - Automatic Ignition Chamber

### PR DESCRIPTION
**Pull request by AtomicScience**
As you know, Smelter, with some probability, extinguish flame under itself. It can annoy some players, and I developed a solution, which can help rich players to craft alloys without being distracted by the constant ignition the block under the Smelter - _Automatic Ignition Chamber_

**How it works:**
In _Guide_ Chamber looks like this:
image

Player places it at the Smelter as an usual Hopper:
image

And loads it with any amounts Flint'n Steels:
image
_Note: For the convenience of players, content of the Chamber won't be output from it (Look new event hander in ItemListener.java)_

So, when player uses the Smelter and flame must be extinguished, Ignition Chamber automatically ignites the block and decrease durability of the loaded Flint'n Stell and play fancy sound of using it.
But, when the durability of the Flint becomes 0, it breaks down (with sound)

And, when Chamber is installed, but no Flints loaded, player'll get this message:
image

### CODE REVIEW

**Comment by AtomicScience** [comment]
Oh, I almost forgot to show the recipe of Chamber.
I tried to make it hard enough, so only rich players can craft it

The recipe is followng:
image

Where:
image - Steel Plate
image - Basic Circuit Board
image - Electromotor
image - Just a Hopper

File: src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java [hunk 1]
```diff
@@ -601,7 +601,7 @@ public boolean onRightClick(ItemUseEvent e, Player p, ItemStack item) {
 			@Override
 			public boolean onRightClick(ItemUseEvent e, Player p, ItemStack item) {
 				if (SlimefunManager.isItemSimiliar(item, SlimefunItems.MAGIC_EYE_OF_ENDER, true)) {
-					e.setCancelled(true);
+					e.getParentEvent().setCancelled(true);
```
Comment (Poslovitch) [inline src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java]: This fix is not related to this PR, therefore it shouldn't have been pushed here. However, as this is already done, let's keep it.

**Comment by TheBusyBiscuit** [comment]
The idea is great.
Haven't looked at the code yet but the lore sounds too much like a 90s tv commercial.
Would be nice if you would maybe make the lore shorter and just more objective about what it does.

**Comment by AtomicScience** [comment]
Okey, I changed lore of the _Chamber_ a little.
Any other ideas?

File: src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java [hunk 2]
```diff
@@ -704,63 +707,101 @@ public boolean onRightClick(ItemUseEvent e, Player p, ItemStack item) {
...
+													resf = (Hopper) raw_disp.getRelative(BlockFace.SOUTH).getState();
+												}
+												boolean isIgniteChamberInstalled = false;
+												
+												if(resf != null) {
+													if(resf.getCustomName().equalsIgnoreCase(SlimefunItems.IGNITION_CHAMBER.getItemMeta().getDisplayName())) isIgniteChamberInstalled = true;
+												}
+												
+												if (SlimefunStartup.chance(100, (Integer) Slimefun.getItemValue("SMELTERY", "chance.fireBreak"))) {
+													if(isIgniteChamberInstalled) {
```
Comment (TheBusyBiscuit) [inline src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java]: Please remove your boolean and use 

BlockStorage.check(block, "IGNITION_CHAMBER") 

instead.

File: src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java [hunk 3]
```diff
@@ -704,63 +707,101 @@ public boolean onRightClick(ItemUseEvent e, Player p, ItemStack item) {
...
+														if(resf.getInventory().contains(Material.FLINT_AND_STEEL)) {
+															ItemStack item = resf.getInventory().getItem(resf.getInventory().first(Material.FLINT_AND_STEEL));
+															item.setDurability((short) (item.getDurability() + 1));
+															if(item.getDurability() >= 64) {
```
Comment (TheBusyBiscuit) [inline src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java]: Use getType().getMaxDurability() instead of 64, just in case Mojang changes it.

File: src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java [hunk 4]
```diff
@@ -704,63 +707,101 @@ public boolean onRightClick(ItemUseEvent e, Player p, ItemStack item) {
...
+																item.setAmount(0); p.getWorld().playSound(p.getLocation(), Sound.ENTITY_ITEM_BREAK, 1, 1);
+																item.setAmount(0);
```
Comment (TheBusyBiscuit) [inline src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java]: item.setAmount(0) is called twice.

File: src/me/mrCookieSlime/Slimefun/listeners/ItemListener.java [hunk 5]
```diff
@@ -58,6 +59,13 @@ public ItemListener(SlimefunStartup plugin) {
 		plugin.getServer().getPluginManager().registerEvents(this, plugin);
 	}
 	
+	@EventHandler
+	public void onIgnitionChamberItemMove(InventoryMoveItemEvent e) {
+		if(e.getInitiator().getName().equalsIgnoreCase(SlimefunItems.IGNITION_CHAMBER.getItemMeta().getDisplayName()) && e.getInitiator().getType() == InventoryType.HOPPER) {
```
Comment (TheBusyBiscuit) [inline src/me/mrCookieSlime/Slimefun/listeners/ItemListener.java]: Just use 

if (BlockStorage.check(block, "IGNITION_CHAMBER")) {

File: src/me/mrCookieSlime/Slimefun/Setup/Messages.java [hunk 6]
```diff
@@ -70,6 +70,7 @@ public static void setup() {
 		local.setDefault("commands.research.reset", "&cYou have reset %player%'s Knowledge");
 		local.setDefault("commands.research.reset-target", "&cYour Knowledge has been reset");
 		local.setDefault("machines.in-use", "&cThis Block's Inventory is currently opened by a different Player.");
+		local.setDefault("machines.ignition-chamber-no-flint", "&cAutomatic Ignition Chamber failed to ignite block because of no Flint'n steel found");
```
Comment (TheBusyBiscuit) [inline src/me/mrCookieSlime/Slimefun/Setup/Messages.java]: Suggested change:

"Ignition Chamber is missing Flint and Steel."

File: src/me/mrCookieSlime/Slimefun/Lists/SlimefunItems.java [hunk 7]
```diff
@@ -361,6 +361,7 @@
 	public static ItemStack GRIND_STONE = new CustomItem(Material.DISPENSER, "&bGrind Stone", 0, new String[] {"", "&a&oGrinds Items for more Efficiency"});
 	public static ItemStack ARMOR_FORGE = new CustomItem(Material.ANVIL, "&6Armor Forge", 0, new String[] {"", "&a&oGives you the Ability to create powerful Armor"});
 	public static ItemStack SMELTERY = new CustomItem(Material.FURNACE, "&6Smeltery", 0, new String[] {"", "&a&oActs as a high-temperature Furnace for Metals"});
+	public static ItemStack IGNITION_CHAMBER = new CustomItem(new ItemStack(Material.HOPPER), "&4Automatic Ignition Chamber", "&rAutomatically ignites the block under the Smelter,", "&ron which it's installed", "&e&oRequires Flint'n Steels to run");
```
Comment (TheBusyBiscuit) [inline src/me/mrCookieSlime/Slimefun/Lists/SlimefunItems.java]: It's Smeltery, not Smelter.
Also, a little misleading as it doesn't ignite the Block. It just prevents it from going out.

"Prevents the Smeltery from using up fire."
"Requires Flint and Steel"
"Must be placed adjacent to the Smeltery's dispenser"

**Comment by TheBusyBiscuit** [comment]
Submitted a code review.

@AtomicScience 
I always love to see new people work on the wiki, as the current team is more or less not doing anything.
However, don't take this personal, but I don't think that your English is suitable for official documentation.
But thank you nonetheless, also a great Block idea.

**Comment by TheBusyBiscuit** [comment]
Another note: Is the Recipe too expensive?

I honestly lost track of balance in Slimefun, don't know if this isn't too much.

File: src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java [hunk 8]
```diff
@@ -704,63 +707,101 @@ public boolean onRightClick(ItemUseEvent e, Player p, ItemStack item) {
...
+											if (InvUtils.fits(inv, adding)) {
+												for (ItemStack removing: inputs.get(i)) {
+													if (removing != null) inv.removeItem(removing);
+												}
+												inv.addItem(adding);
+												p.getWorld().playSound(p.getLocation(), Sound.BLOCK_LAVA_POP, 1, 1);
+												p.getWorld().playEffect(b.getLocation(), Effect.MOBSPAWNER_FLAMES, 1);
+												Block raw_disp = b.getRelative(BlockFace.DOWN);									
+												Hopper resf = null;
+												boolean isIgnitionChamberInstalled = false;
```
Comment (TheBusyBiscuit) [inline src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java]: You could just remove this boolean and check whether resf != null instead.

Comment (TheBusyBiscuit) [reply src/me/mrCookieSlime/Slimefun/Setup/SlimefunSetup.java]: Also why is it named "resf"?

**Comment by TheBusyBiscuit** [comment]
I would rather change the variable to 'chamber' 'resf' is a bit confusing.
After that, it should be ready to merge.
